src/beamlime/applications/interfaces.py

```diff
@@ -1,195 +1,35 @@
-
-
-class ApplicationPausedException(Exception):
-    ...
-
-
-class ApplicationStoppedException(Exception):
-    ...
-
-
-class _DaemonInterface(ABC):
+class BeamlimeApplicationInterface(LogMixin, FlagControlMixin, CoroutineMixin, ABC):
     """
-    Daemon Application Interface
+    Beamlime Application Interface
 
+    Protocol
+    --------
+    BeamlimeApplicationProtocol
     """
 
-    def __del__(self):
-        self.stop()
-        if hasattr(super(), "__del__"):
-            super().__del__()
-
-    async def stopped(
-        self,
-        timeout: int = MAX_INSTANCE_PAUSED,
-        check_pause_interval: int = MAX_INSTANCE_PAUSED,
-    ) -> bool:
-        @async_timeout(ApplicationPausedException)
-        async def wait_resumed(timeout: int, wait_interval: int):
-            if self._stopped:
-                self.info("Application stopped.")
-                return False
-            elif not self._paused:
-                return True
-            elif self._paused:
-                self.info(
-                    "Application process paused. Waiting for ``resume`` command..."
-                )
-                raise ApplicationPausedException
-
-        try:
-            resumed = await wait_resumed(timeout, check_pause_interval)
-            return not resumed
-        except TimeoutError:
-            self.info("Application paused limit reached. Stopping the application...")
-            self.stop()
-            return True
-
-    @abstractmethod
-    async def _run(self) -> None:
-        """
-        Application coroutine generator.
-        """
-        ...
-
-    def run(self) -> None:
-        """
-        Run the application in a dependent event loop.
-
-        """
-        # TODO: ``run`` should also check if the communication-interface
-        # is multi-process-friendly before running.
-        # For example, if one of input_channel or output_channel is ``Queue``,
-        # the application daemon should start by ``create_task`` not, ``run``.
-
-        return asyncio.run(self._run())
-
-    def create_task(self, /, name=None, context=None) -> asyncio.Task:
-        """
-        Start the application task in the currently running event loop.
-        """
-        if not hasattr(self, "_task") or self._task.done():
-            # TODO: Remove try-except after updating the minimum python version to 3.11
-            try:
-                self._task = asyncio.create_task(
-                    self._run(), name=name, context=context
-                )  # py311
-            except TypeError:
-                self._task = asyncio.create_task(self._run(), name=name)  # py39, py310
-        return self._task
-
-
-class BeamlimeApplicationInterface(_LogMixin, _FlagControlMixin, _DaemonInterface, ABC):
     def __init__(self, config: dict = None, logger=None, **kwargs) -> None:
         self._pause_interval = 0.1
         self._init_logger(logger=logger)
         self.app_name = kwargs.get("name", "")
```

Comment:
Why use `get` on `kwargs` here (and see below) and not have function arguments with default values (which would also make generating API docs more easy)? Is there a technical reason I am missing?

Reply:
It was only because I was not sure where those variables should be set, either `__init__` or `parse_config` and the `parse_config` ... 
and it was easier to keep this form in case I want them to be parsed from `config` : D.... 
Still not sure how should they be, either
1. `ApplicationGroup` parses the configuration and create instance accordingly,
2. `handler` constructor should have all the key-word arguments with default values that is expecting from the configuration so that `BeamlimeSystem` can simply pass the subset of configuration...? 
I'll figure it out as soon as possible...